docker/Dockerfile

```diff
@@ -31,12 +33,15 @@ COPY setup/setup_pddlstream.bash setup/
 RUN setup/setup_pddlstream.bash
 
 # Install pip dependencies for testing
+ARG VIRTUAL_ENV
+RUN python3 -m venv ${VIRTUAL_ENV}
+ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
```

Comment:
I'm learning a lot about venvs from you! In this case I might need a little more information - since it feels very "un-docker-y" to use a venv (I don't know if that's true). Would it be preferable to add a test target after installing baseline dependencies in the `pyrobosim` image? Is there some specific advantage to a venv?

Reply:
There are some weird issues that I run into building the `base` image... definitely unrelated here but man pip is messing me up.

```
8.903 Collecting PyQt5 (from pyrobosim==1.2.0)
8.931   Downloading PyQt5-5.15.10.tar.gz (3.2 MB)
11.68      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 1.2 MB/s eta 0:00:00
11.95   Installing build dependencies: started
17.11   Installing build dependencies: finished with status 'done'
17.11   Getting requirements to build wheel: started
17.18   Getting requirements to build wheel: finished with status 'done'
17.18   Preparing metadata (pyproject.toml): started
17.26   Preparing metadata (pyproject.toml): finished with status 'error'
17.26   error: subprocess-exited-with-error
17.26   
17.26   × Preparing metadata (pyproject.toml) did not run successfully.
17.26   │ exit code: 1
17.26   ╰─> [23 lines of output]
17.26       pyproject.toml: line 7: using '[tool.sip.metadata]' to specify the project metadata is deprecated and will be removed in SIP v7.0.0, use '[project]' instead
```

Reply:
So in Ubuntu 24.04 (or Python 3.12?) you get an error by default if you directly try install packages into system Python.

The 2 ways to get around that are:
1. Make a virtual environment
2. Add the `--break-system-packages` arg (which does not exist as an option in older versions)

Neither of them is ideal, but the first one can be made to work in both older and newer versions... at some cost as you can see here.

Reply:
As for your error, I wonder if the version of PyQt5 released for your architecture is wonky. See if you can install a different version than whatever is pinned, or maybe pin it to an older version if not set in the requirements?

Reply:
Actually, I found a much better solution!

```
ENV PIP_BREAK_SYSTEM_PACKAGES=1
```

Reply:
> pyproject.toml: line 7: using '[tool.sip.metadata]' to specify the project metadata is deprecated and will be removed in SIP v7.0.0, use '[project]' instead

This one seems like a known issue... sigh
https://github.com/altendky/pyqt-tools/issues/120

Reply:
I guess this is now moot because of your migration off PyQt5... but for the build issues I ran into I just needed to `apt-get install -y qtbase5-dev` prior to installing the Python package. I guess you need to have the [build tools installed](https://pypi.org/project/PyQt5/) and for some reason in my docker build they just weren't... Also needing to somehow auto accept the GPL license terms.

Either way if you merge the migration all those issues go away and I can have that hour of my life back.